packages/vm/src/cache.rs

```diff
@@ -139,10 +139,9 @@ where
             return self.pinned_memory_cache.store(checksum, module);
         }
 
-        // Load code from the file system
+        // Re-compile from original Wasm bytecode
         let code = self.load_wasm(checksum)?;
-        // Compile and store into the pinned cache
-        let module = compile_only(code.as_slice())?;
+        let module = compile_and_use(&code, Some(self.instance_memory_limit))?;
```

Comment:
Good fix. I've missed that, with the "compile" vs. "use" distinction.

Reply:
No worries. If you have naming suggestions to make this clearer, let me know. But modules from `compile_only` should not be used directly.

Reply:
> If you have naming suggestions to make this clearer, let me know.

I was thinking exactly about that. What about just a `compile` method, with an `Option<Size>` parameter for `memory_limit`? So that you "stumble" with the parameter when trying to use the method.

> But modules from `compile_only` should not be used directly.

I think the only way to try to "enforce" that would be through documentation. In that regard, having a single method may help rather than hurt.

Reply:
Created #727 to implement this.